File: web/console/src/features/tasks/agentWorkflow.ts
/**
 * 功能：定义 Agent 工作流表单与摘要共用的安全引用边界。
 * 实现：按后端的长度、字符和凭据前缀规则过滤；不读取任何目标配置。
 * 输入：未知目录或响应字段；输出：安全 ID 或 undefined；依赖：无。
 */
const CREDENTIAL_PREFIX = /^(sk[-_]|ghp_|github_pat_|akia|asia|eyj|bearer |token |password)/i

export function safeAgentReference(value: unknown): string | undefined {
  return typeof value === 'string' && /^[A-Za-z0-9][A-Za-z0-9._ -]{0,127}$/.test(value)
    && !value.includes('..') && !CREDENTIAL_PREFIX.test(value) ? value : undefined
}

export function safeEvaluationModelReference(value: unknown): string | undefined {
  return typeof value === 'string' && /^[A-Za-z0-9][A-Za-z0-9._:-]{0,127}$/.test(value)
    && !CREDENTIAL_PREFIX.test(value) ? value : undefined
}

export function safeReportReference(value: unknown): string | undefined {
  return typeof value === 'string' && /^[A-Za-z0-9][A-Za-z0-9_-]{0,127}$/.test(value)
    && !CREDENTIAL_PREFIX.test(value) ? value : undefined
}

File: internal/platform/tasks/agent_workflow.go
package tasks

import (
	"encoding/json"
	"regexp"
	"strings"
	"unicode/utf8"
)

var safeAgentIDPattern = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9._ -]{0,127}$`)

// validNewAgentInput 只约束新任务；历史幂等确认在调用前已经完成。
func validNewAgentInput(input CreateInput) bool {
	return strings.TrimSpace(input.Content) != "" && utf8.ValidString(input.Content) && len(input.AttachmentIDs) == 0
}

// safeAgentReferences 仅投影完整标准引用合同，不回显历史内联配置或疑似凭据。
func safeAgentReferences(raw json.RawMessage) (string, string) {
	var params agentTaskParams
	if !validTaskParams("agent_scan", raw) || !decodeExactJSON(raw, &params) {
		return "", ""
	}
	agentID, modelID := params.AgentID, params.EvalModelID
	if !safeAgentIDPattern.MatchString(agentID) || strings.Contains(agentID, "..") || suspiciousReference(agentID) {
		agentID = ""
	}
	if safeModelID(modelID) == "" || suspiciousReference(modelID) {
		modelID = ""
	}
	return agentID, modelID
}

func suspiciousReference(value string) bool {
	lower := strings.ToLower(value)
	for _, prefix := range []string{"sk-", "sk_", "ghp_", "github_pat_", "akia", "asia", "eyj", "bearer ", "token ", "password"} {
		if strings.HasPrefix(lower, prefix) {
			return true
		}
	}
	return false
}
